src/calls/embedTemplatePage.ts

```diff
@@ -121,3 +111,98 @@ function createEmbedIframe(options: CreateIframeOptions): HTMLIFrameElement {
 
   return iframe;
 }
+
+function parseTokenPayload(token: string): SdkTokenPayload {
+  const tokenPayloadResult = sdkTokenPayloadSchema.safeParse(
+    JSON.parse(atob(token.split(".")[1] ?? "")),
+  );
+
+  if (!tokenPayloadResult.success) {
+    throw new Error("Invalid token payload");
+  }
+
+  return tokenPayloadResult.data;
+}
+
+function validatePermissions(
+  permissions: SdkPermission[],
+  features?: TemplateEmbedFeatures,
```

Comment:
Could default here to skip the check later on.
```suggestion
  features: TemplateEmbedFeatures = {},
```